src/backend/distributed/test/shard_rebalancer.c

```diff
@@ -500,48 +471,63 @@ JsonFieldValueUInt32(Datum jsonDocument, const char *key)
  * document and returns it as an unsigned 64-bit integer.
  */
 static uint64
-JsonFieldValueUInt64(Datum jsonDocument, const char *key)
+JsonFieldValueUInt64(Datum jsonDocument, const char *key, uint64 fallback)
 {
 	char *valueString = JsonFieldValueString(jsonDocument, key);
+	if (valueString == NULL)
+	{
+		return fallback;
+	}
 	Datum valueInt8Datum = DirectFunctionCall1(int8in, CStringGetDatum(valueString));
 
 	uint64 valueUInt64 = DatumGetInt64(valueInt8Datum);
 	return valueUInt64;
 }
 
 
+/*
+ * DirectFunctionalCall2Null is a version of DirectFunctionCall2 that can
+ * return NULL. It still does not support NULL arguments though.
+ */
+static Datum
+DirectFunctionCall2Null(PGFunction func, bool *isnull, Datum arg1, Datum arg2)
```

Comment:
Do we have a file where we store functions that we copied from postgres? This might fit better there.